### PR TITLE
DAT-11471: fixed issue with generation mssql create view/procedure sql statement when source DDL has CREATE OR ALTER statement

### PR DESCRIPTION
**Pull request by achehrynets**

### CODE REVIEW

**Review by v-petrovych — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java [hunk 1]
```diff
@@ -70,15 +70,11 @@ public Sql[] generateSql(CreateProcedureStatement statement, Database database,
...
             sql.add(new UnparsedSql("if object_id('" + fullyQualifiedName + "', 'p') is null exec ('create procedure " + fullyQualifiedName + " as select 1 a')"));
 
-            StringClauses parsedSql = SqlParser.parse(procedureText, true, true);
-            StringClauses.ClauseIterator clauseIterator = parsedSql.getClauseIterator();
-            Object next = "START";
-            while (next != null && !(next.toString().equalsIgnoreCase("create") || next.toString().equalsIgnoreCase("alter")) && clauseIterator.hasNext()) {
-                next = clauseIterator.nextNonWhitespace();
+            StringClauses parsedProcedureDefinition = SqlParser.parse(procedureText, true, true);
+            if (parsedProcedureDefinition.contains("CREATE") && !(parsedProcedureDefinition.contains("OR") && parsedProcedureDefinition.contains("ALTER"))) {
+                parsedProcedureDefinition.replace("CREATE", "ALTER");
```
Comment (andrulban) [inline liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java:75]: @achehrynets what if statement has multiple "create" words in it, for example as part of procedure body, is it possible? If it is, won't this line replace all the matches and not only the first one?

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateViewGenerator.java [hunk 2]
```diff
@@ -94,7 +94,9 @@ public Sql[] generateSql(CreateViewStatement statement, Database database, SqlGe
                 CatalogAndSchema schema = new CatalogAndSchema(statement.getCatalogName(), statement.getSchemaName()).customize(database);
                 sql.add(new UnparsedSql("IF NOT EXISTS (SELECT * FROM sys.views WHERE object_id = OBJECT_ID(N'[" + schema.getSchemaName() + "].[" + statement.getViewName() + "]'))\n" +
                         "    EXEC sp_executesql N'CREATE VIEW [" + schema.getSchemaName() + "].[" + statement.getViewName() + "] AS SELECT ''This is a code stub which will be replaced by an Alter Statement'' as [code_stub]'"));
-                viewDefinition.replaceIfExists("CREATE", "ALTER");
+                if (viewDefinition.contains("CREATE") && (!viewDefinition.contains("OR") || !viewDefinition.contains("ALTER"))) {
+                    viewDefinition.replace("CREATE", "ALTER");
```
Comment (andrulban) [inline liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateViewGenerator.java:98]: Same here

**Review by andrulban — CHANGES_REQUESTED** [review]

**Review by andrulban — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java (hunk 1 above)
Comment (achehrynets) [reply liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java:75]: I added unit tests and found one possible problem which could happen and fixed it

File: liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateViewGenerator.java (hunk 2 above)
Comment (achehrynets) [reply liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateViewGenerator.java:98]: I added unit tests and found one possible problem which could happen and fixed it

**Review by rsian-lb — APPROVED** [review]

**Review by v-petrovych — APPROVED** [review]

**Review by andrulban — APPROVED** [review]

**Review by v-petrovych — APPROVED** [review]

**Pull request merged**